### PR TITLE
sql/parser: support STRING alias for varchar

### LINKED ISSUE #5892: STRING(n) isn't supported
When you declare a column as `CHAR(n)` or `VARCHAR(n)`, we convert that to `STRING(n)`:

```
$ create table strings (a CHAR(4), b VARCHAR(4));
CREATE TABLE
$ show columns from strings;
+-------+-----------+-------+----------------+
| Field |   Type    | Null  |    Default     |
+-------+-----------+-------+----------------+
| a     | STRING(4) | true  | NULL           |
| b     | STRING(4) | true  | NULL           |
| rowid | INT       | false | unique_rowid() |
+-------+-----------+-------+----------------+
```

But you can't currently declare a column as `STRING(n)`:

```
$ create table strings (a STRING(4), b VARCHAR(4));
pq: syntax error at or near "("
create table strings (a STRING(4), b VARCHAR(4));
                              ^
```

@petermattis

### PR DESCRIPTION
**Pull request by dt**
Fixes #5892

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/5918)

### CODE REVIEW

**Comment by RaduBerinde** [comment]
:lgtm:

---

Reviewed 4 of 4 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/5918#-:-KEkwnq7baYF3JrhaZng:1417088721)*

File: sql/parser/sql.y [hunk 1]
```diff
@@ -2702,6 +2702,10 @@ character_base:
   {
     $$.val = &StringType{Name: "VARCHAR"}
   }
+| STRING
+  {
+    $$.val = &StringType{Name: "STRING"}
+  }
```
Comment (petermattis) [inline sql/parser/sql.y]: I think you can remove `STRING` from the `simple_typename` rule.

File: sql/testdata/table [hunk 2]
```diff
@@ -138,17 +138,23 @@ CREATE TABLE test.users (
   id    INT PRIMARY KEY,
   name  VARCHAR NOT NULL,
   title VARCHAR,
+  nickname STRING,
+  username STRING(10),
+  email VARCHAR(100),
```
Comment (petermattis) [inline sql/testdata/table]: My OCD is tingling now that the fields are no longer aligned.

**Comment by petermattis** [comment]
LGTM

File: sql/testdata/table (hunk 2 above)
Comment (dt) [inline sql/testdata/table]: Done.

File: sql/parser/sql.y (hunk 1 above)
Comment (dt) [inline sql/parser/sql.y]: Done.

**Comment by jseldess** [comment]
Updated docs with https://github.com/cockroachdb/docs/pull/211

**Pull request merged**